マークアップ/トップページのビュー作成

GotoUkyo opened the pull request:
# What
html及びsassを使用し、フリマアプリのトップページのビューを作成
それに伴うデータベースの作成、コントローラーの作成、gemの追加、ダウンロード素材の導入

# Why
トップページはアプリを使用する際、最初に目にするページであり、顔となるページでもある。
その為分かりやすさ（イラストを使ったアプリ説明や文字の装飾による強調等）とデザイン性（見た目の華やかさ）を意識して作成することは、使用時は勿論、これから登録しようかというユーザーへのアピールポイントにもなる。

GotoUkyo committed:
[追加]Gemfile

hamlとFont Awesomeを実装するために必要なgem
・haml-rails
・font-awesome-sass
を追加し、bundle installを実行

GotoUkyo committed:
[追加]ビュー作成のための準備

・トップページ（index）を表示するためのコントローラーを作成
　→indexes_controller.rb
・トップページのhamlコードを書き込むファイル
　index.html.haml
　_header.html.haml
　_main.html.haml
を作成し、index.html.hamlは下二つを表示するよう設定

GotoUkyo committed:
[作成]トップページのhamlとscss

個人作業で作っていたコード
・_header.html.haml
・_main.htmla.haml
・indexes.scss
上記3ファイルの記述をコピー＆ペースト

GotoUkyo committed:
[追加]app/assets/images

ダウンロードした画像・ロゴ素材をimagesファイルに配置

GotoUkyo committed:
[追加]application.css

・リセットscssを追加し、設定
・font-awesomeが適用
・index.html.hamlにscssが適用されるように設定

GotoUkyo committed:
erbファイルをhaml形式に変換

ターミナルにてhaml:erb2hamlを実行

GotoUkyo committed:
[編集]indexes/_main.html.haml

ページ上部の画像（.Top-img）を設定

GotoUkyo committed:
[編集]indexes/_main.html.haml

ページ中央の画像（.Center_img）を設定

GotoUkyo committed:
[編集]indexes/_main.html.haml

ページ下部の画像（.Under-img）を設定

GotoUkyo committed:
[編集]indexes/_header.html.haml

ページ左上のロゴ（.logo）を設定

GotoUkyo committed:
画像3点の実装

indexes/_main.html.haml
modules/_indexes.scss
上記ファイルを編集して「FURIMAが選ばれる3つの理由」部分に画像を3点実装

GotoUkyo committed:
画像上に四角形＋数字を追加

indexes/_main.html.haml
modules/_indexes.scss
上記ファイルを編集し、「フリマが選ばれる3つの理由」部分の画像左上にそれぞれ数字が振られた四角形を設定

GotoUkyo committed:
四角形の大きさを調整

少し小さかったので_indexes.scssを編集し大きさを変更

GotoUkyo committed:
[編集]アイコン3点の実装

indexes/_main.html.haml
modules/_indexes.scss
・上記2ファイルを編集し、「FURIMAの特徴」部分にアイコン3点を実装
・それに伴い要素の大きさや間隔を調整

GotoUkyo committed:
[追加]ページ下部にロゴを追加

indexes/_main.html.haml
modules/_indexes.scss
上記2ファイルを編集し、ページ最下部（Footer内最下部）にロゴとコピーライトを追加

GotoUkyo committed:
[編集]検索ボタンにアイコンを表示

indexes/_header.html.haml
modules/_index.scss
上記2ファイルを編集し、検索バー横のボタンに虫眼鏡アイコンを設定

GotoUkyo committed:
[編集]検索タイトル前にアイコンを表示

indexes/_header.html.haml
modules/_indexes.scss
ヘッダーの「カテゴリから探す」「ブランドから探す」の前にそれぞれのアイコンを設定

GotoUkyo committed:
[追加]indexes/_header.html.haml

・ヘッダーのフォルダ（indexes/_header.html.haml）及びmodules/_indexes.scssを編集し、画面右下に「出品する」ボタンが固定で表示されるよう設定
・それに伴うテキスト、アイコンの設定

GotoUkyo committed:
[削除]modules/_indexes.scss

不要になったbackground-colorの削除

GotoUkyo committed:
[編集]modules/_indexes.scss

背景色・アイコンサイズ・文字の位置等細かい修正

daisukekamakura commented:
LGTMです。

GotoUkyo committed:
Font awesomeに関する項目の削除

Font awesomeを使わなかったので、関連するコードを下記ファイルから削除（bundle install済）
・stylesheets/application.css
・Gemfile

yupimaru commented on app/assets/stylesheets/modules/_indexes.scss:
```diff
@@ -0,0 +1,695 @@
+* {
```
1つのファイルに695行は多すぎます。
可読性も悪く、メンテナンスもしにくくなります。

ファイルを切り分けましょう。

GotoUkyo replied:
_indexes.scssを_header.scssと_main.scssに分けました。

yupimaru commented on app/assets/stylesheets/modules/_indexes.scss:
```diff
@@ -0,0 +1,695 @@
+        }
+      }
+    }
+  }
```
SCSSにおけるネストは3つ程度が妥当です。
今回の修正は不要ですが、次回以降意識しましょう。

GotoUkyo replied:
覚えて今後意識しておきます。ありがとうございます。

yupimaru commented on app/views/indexes/_header.html.haml:
```diff
@@ -0,0 +1,32 @@
+.body-header
+  .ichidan
```
```haml
.ichidan
```

こちらのクラス名は適切でしょうか？再考しましょう。

GotoUkyo replied:
headerの上の段、ということで.upという名前に変更しました。

yupimaru commented on app/views/indexes/_header.html.haml:
```diff
@@ -0,0 +1,32 @@
+.body-header
+  .ichidan
+    .logo
+      %img{src: "assets/logo/logo.png", class: "logo__img"}/
```
```haml
%img{src: "assets/logo/logo.png", class: "logo__img"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/logo/logo.png", class: "logo__img"に書き換えました。

yupimaru commented on app/views/indexes/_header.html.haml:
```diff
@@ -0,0 +1,32 @@
+      .Search-bar__inputBox
+        %input.Search-bar__inputBox--text{placeholder: "キーワードから探す", "type": "text"}
+        .Search-bar__inputBox--btn
+          %img{src: "assets/icon/icon-search_1.png", class: "Search-bar__inputBox--btn__icon"}/
```
```haml
%img{src: "assets/icon/icon-search_1.png", class: "Search-bar__inputBox--btn__icon"}/
```
imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/icon/icon-search_1.png", class: "Search-bar__inputBox--btn__icon"に書き換えました。

yupimaru commented on app/views/indexes/_header.html.haml:
```diff
@@ -0,0 +1,32 @@
+    .Search-menu
+      .Search-menu__category
+        = link_to "", class: "Search-menu__category--btn" do
+          %img{src: "assets/icon/icon_category.png", class: "Search-menu__category--btn__icon"}/
```
```haml
%img{src: "assets/icon/icon_category.png", class: "Search-menu__category--btn__icon"}/
```
imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/icon/icon_category.png", class: "Search-menu__category--btn__icon"に書き換えました。

yupimaru commented on app/views/indexes/_header.html.haml:
```diff
@@ -0,0 +1,32 @@
+          カテゴリーから探す
+      .Search-menu__brand
+        = link_to "", class: "Search-menu__brand--btn" do
+          %img{src: "assets/icon/icon_brand.png", class: "Search-menu__brand--btn__icon"}/
```
```haml
%img{src: "assets/icon/icon_brand.png", class: "Search-menu__brand--btn__icon"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/icon/icon_brand.png", class: "Search-menu__brand--btn__icon"に書き換えました。

yupimaru commented on app/views/indexes/_header.html.haml:
```diff
@@ -0,0 +1,32 @@
+  .Exhibit
+    .Exhibit__text 出品する
+    .Exhibit__icon
+      %img{src: "assets/icon/icon_camera.png", class: "Exhibit__icon--pict"}/
```
```haml
%img{src: "assets/icon/icon_camera.png", class: "Exhibit__icon--pict"}/ 
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/icon/icon_camera.png", class: "Exhibit__icon--pict"に書き換えました。

yupimaru commented on app/views/indexes/_header.html.haml:
```diff
@@ -0,0 +1,32 @@
+        %input.Search-bar__inputBox--text{placeholder: "キーワードから探す", "type": "text"}
+        .Search-bar__inputBox--btn
+          %img{src: "assets/icon/icon-search_1.png", class: "Search-bar__inputBox--btn__icon"}/
+  .nidan
```
```haml
.nidan
```

こちらのクラス名は適切でしょうか？再考しましょう。

GotoUkyo replied:
headerの下の段、ということで.underという名前に変更しました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+.body-main
+  .Top-img
+    %img{src: "assets/pict/bg-mainVisual-pict_pc.jpg", class: "Top-img__pict"}/
```
```haml
%img{src: "assets/pict/bg-mainVisual-pict_pc.jpg", class: "Top-img__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/pict/bg-mainVisual-pict_pc.jpg", class: "Top-img__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+      .Reason__top--ber
+    .Reason__img
+      .Reason__img--left
+        %img{src: "assets/pict/pict-reason-01.jpg", class: "Reason__img--left__pict"}/
```
```haml
%img{src: "assets/pict/pict-reason-01.jpg", class: "Reason__img--left__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/pict/pict-reason-01.jpg", class: "Reason__img--left__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+        .Number
+          .Number__one 1
+      .Reason__img--center
+        %img{src: "assets/pict/pict-reason-02.jpg", class: "Reason__img--center__pict"}/
```
```haml
%img{src: "assets/pict/pict-reason-02.jpg", class: "Reason__img--center__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/pict/pict-reason-02.jpg", class: "Reason__img--center__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+        .Number
+          .Number__two 2
+      .Reason__img--right
+        %img{src: "assets/pict/pict-reason-03.jpg", class: "Reason__img--right__pict"}/
```
```haml
%img{src: "assets/pict/pict-reason-03.jpg", class: "Reason__img--right__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/pict/pict-reason-03.jpg", class: "Reason__img--right__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+      .Reason__text--right
+        10%でお得に出品&購入！
+  .Center-img
+    %img{src: "assets/pict/bg-topMiddleDl-pict.jpg", class: "Center-img__pict"}/
```
```haml
%img{src: "assets/pict/bg-topMiddleDl-pict.jpg", class: "Center-img__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/pict/bg-topMiddleDl-pict.jpg", class: "Center-img__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+      .Feature__top--ber
+    .Feature__img
+      .Feature__img--left
+        %img{src: "assets/icon/icon-01.png", class: "Feature__img--left__pict"}/
```
```haml
%img{src: "assets/icon/icon-01.png", class: "Feature__img--left__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/icon/icon-01.png", class: "Feature__img--left__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+      .Feature__img--left
+        %img{src: "assets/icon/icon-01.png", class: "Feature__img--left__pict"}/
+      .Feature__img--center
+        %img{src: "assets/icon/icon-03.png", class: "Feature__img--center__pict"}/
```
```haml
%img{src: "assets/icon/icon-03.png", class: "Feature__img--center__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/icon/icon-03.png", class: "Feature__img--center__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+      .Feature__img--center
+        %img{src: "assets/icon/icon-03.png", class: "Feature__img--center__pict"}/
+      .Feature__img--right
+        %img{src: "assets/icon/icon-04.png", class: "Feature__img--right__pict"}/
```
```haml
%img{src: "assets/icon/icon-04.png", class: "Feature__img--right__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/icon/icon-04.png", class: "Feature__img--right__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+      .PickBrand__img--new
+      .PickBrand__img--new
+  .Under-img
+    %img{src: "assets/pict/bg-appBanner-pict.jpg", class: "Under-img__pict"}/
```
```haml
%img{src: "assets/pict/bg-appBanner-pict.jpg", class: "Under-img__pict"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/pict/bg-appBanner-pict.jpg", class: "Under-img__pict"に書き換えました。

yupimaru commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+          %p お知らせ
+    .Footer__logo
+      .Footer__logo--img
+        %img{src: "assets/logo/logo-white.png", class: "Footer__logo--img"}/
```
```haml
%img{src: "assets/logo/logo-white.png", class: "Footer__logo--img"}/
```

imgタグは、直接タグを記述するのではなく、Railsのヘルパーメソッドを用いて記載しましょう。
もしわからなければ、カリキュラムを復習したり、Railsの公式リファレンスを確認しましょう。

GotoUkyo replied:
該当部分を= image_tag src="/assets/logo/logo-white.png", class: "Footer__logo--img"に書き換えました。

yupimaru commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,11 @@
+!!!
+%html
+  %head
+    %meta{:content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
```
```haml
%meta{:content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
```

ハッシュロケット記法は古い記法です。シンボル型へ修正しましょう。

GotoUkyo replied:
該当部分をシンボル型に書き換えました。

yupimaru commented on app/assets/stylesheets/modules/_indexes.scss:
```diff
@@ -0,0 +1,695 @@
+        color: white;
+        text-decoration: none;
+        border-radius: 5px;
+        background-color: lightseagreen;
```
```scss
background-color: lightseagreen;
```

頻繁に使用する色は、変数化して使用しましょう。
他に頻繁に使用する色があれば同様です。

GotoUkyo replied:
lightseagreenとwhiteの2色を変数化し、_header.scssと_main.scssで使用しました。

yupimaru reviewed (commented):
レビュー依頼ありがとうございます。

コメントしておりますので、ご確認のほどよろしくお願いいたします。

また、修正後再提出される際には、現在ブラウザでどのように表示されているのか確認させていただきたいため、Gyazoを用いてスクリーンショットを撮影して頂き、生成されたURLを当プルリクエストへ添付して下さい。

GotoUkyo committed:
[修正]cssファイルを2つに

・1つに纏めていた_header.html.hamlと_main.html.haml用のcssファイルをそれぞれ1つずつに分割
・それに伴いstylesheets/application.cssを編集

GotoUkyo committed:
[編集]一部クラス名を変更

headerのクラス.ichidanと.nidanをそれぞれ.upと.underに変更

GotoUkyo committed:
[編集]_header.html.haml

%img{}をimage_tagに変更

GotoUkyo committed:
[編集]_main.html.haml

%img{}をimage_tagに変更

GotoUkyo committed:
[修正]layouts/application.html.haml

ハッシュロケット記法の記述をシンボル型に修正

GotoUkyo committed:
[編集]色を変数化

modules/_header.scss
modules/_main.scss
上記ファイルで2色変数化
lightseagreen→--main-color
white→--white-color

GotoUkyo commented:
https://gyazo.com/142e237c4294bd530b79a4ef1489df7e
こちら、作成したページを当方のブラウザ上で表示し、スクロールしているgif画像になります。
再度レビューの程、何卒よろしくお願いいたします。

syuji-yamamoto commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+        %p フリマアプリです
+      .Top-img__in--icon
+        .icon-link
+        .icon-link.icon-link__2
```
見本サイトとAppstoreとGooglestoreの画像を表示できるようにしましょう。
素材は検索で探してみてください。

GotoUkyo replied:
.icon-linkにAppstoreのアイコンと仮リンクを設定しました。

syuji-yamamoto commented on app/views/indexes/_main.html.haml:
```diff
@@ -0,0 +1,166 @@
+        今すぐ無料ダウンロード！
+      .Under-img__in--icon
+        .icon-link
+        .icon-link.icon-link__2
```
こちらも同様です。
見本サイトとAppstoreとGooglestoreの画像を表示できるようにしましょう。
素材は検索で探してみてみてください。

GotoUkyo replied:
.icon-link2にGooglestoreのアイコンと仮リンクを設定しました。

syuji-yamamoto commented on app/views/indexes/_header.html.haml:
```diff
@@ -0,0 +1,32 @@
+  .under
+    .Search-menu
+      .Search-menu__category
+        = link_to "", class: "Search-menu__category--btn" do
```
リンク先は"#"にしておくと良いでしょう。```link_to "#"```
""(空のリンク)は思わぬエラーを起こす可能性があります。
他の箇所も同様です。

GotoUkyo replied:
リンクの設定先として#を仮挿入しました。

syuji-yamamoto reviewed (commented):
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
数点、追加でコメントしましたので、ご確認をお願いします！

GotoUkyo committed:
[修正]リンク設定の修正

・空になっているリンク設定に#を仮挿入
・右下の「出品する」アイコンにリンクを設定し忘れていたので同様に設定

GotoUkyo committed:
[編集]AppstoreとGooglestoreアイコンの設定

_main.html.haml
_main.scss
上記2ファイルを編集し、黒いスペースを仮置きしていた部分にAppstoreとGooglestoreのアイコンと仮リンクを設定

GotoUkyo committed:
[修正]_main.scss

構造が一部乱れていたので見やすいように修正

GotoUkyo committed:
[修正]一部クラス名を修正

分かりにくいと感じたのでクラス名を以下のように修正
icon-link　→　icon-app
icon-link2　→　icon-google

GotoUkyo commented:
分かりにくいと感じたので以下のように一部クラス名を変更しました。
icon-link　→　icon-app
icon-link2　→　icon-google

修正後のページをブラウザ上で表示し、スクロールした画面のgif画像がこちらです。
https://gyazo.com/62dc6133cf633d679931d68c6d91d3f1

再度レビューの程、何卒よろしくお願いいたします。

syuji-yamamoto commented:
レビュー依頼ありがとうございます！
修正点を確認させていただきました！
LGTMとさせていただきます！実装お疲れ様でした！
マージして本番環境にデプロイ後、確認をお願いします！
引き続き、実装頑張っていきましょう！

GotoUkyo commented:
レビューしてくださりありがとうございました！
以降の作業に進ませていただきます！
